src/middlewareV2/registrar/modules/SocketRegistry.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import {ISocketRegistry} from "../../../interfaces/ISocketRegistryV2.sol";
+import {SocketRegistryStorage} from "./SocketRegistryStorage.sol";
+import {
+    OperatorSetLib,
+    OperatorSet
+} from "eigenlayer-contracts/src/contracts/libraries/OperatorSetLib.sol";
+
+/// @notice A module that allows for the setting and removal of operator sockets
+/// @dev This contract assumes a single socket per operator
+abstract contract SocketRegistry is SocketRegistryStorage {
+    using OperatorSetLib for OperatorSet;
+
+    /// @inheritdoc ISocketRegistry
+    function getOperatorSocket(
+        address operator
+    ) external view returns (string memory) {
+        return _operatorToSocket[operator];
+    }
+
+    /// @inheritdoc ISocketRegistry
+    function updateSocket(address operator, string memory socket) external {
+        require(msg.sender == operator, CallerNotOperator());
```

Comment:
Should there be a check to see if the operator is registerd to atleast one operatorSet of the AVS?